src/main/java/com/example/demo/controller/F1DriverStatsController.java

```diff
@@ -0,0 +1,144 @@
+        Driver savedDriver = driverRepository.save(driver);
+        savedDriver.setCurrentTeam(Team.builder().teamName("Ferrari").build());
+        return new ResponseEntity<>(savedDriver, HttpStatus.CREATED);
+    }
+
+    @GetMapping(path = "/all")
+    public Iterable<Driver> getAllDrivers() {
+        return driverRepository.findAll();
+    }
+
+    @GetMapping(path = "/{id}")
+    public Optional<Driver> getDriverById(@PathVariable Long id) {
+        return driverRepository.findById(id);
+    }
+
+    @GetMapping(path = "/get-all-firstname-{firstName}")
+    public Iterable<Driver> getDriversWithFirstName(@PathVariable String firstName) {
+        return driverRepository.findAllByFirstName(firstName);
+    }
+
+    @GetMapping(path = "/get-all-lastname-{lastName}")
+    public Iterable<Driver> getDriversWithLastName(@PathVariable String lastName) {
+        return driverRepository.findAllByLastName(lastName);
+    }
+
+    @GetMapping(path = "/most-wins")
+    public Iterable<Driver> getDriverWithMostWins() {
+        return driverRepository.findFirstByOrderByWinsDesc();
+    }
+
+    @GetMapping(path = "/wins")
+    public @ResponseBody
+    Iterable<Driver> winsBetween(@RequestParam int from, @RequestParam int to) {
+        return driverRepository.findDriversByWinsBetween(from, to);
+    }
+
+    @GetMapping(path = "/most-points")
+    public @ResponseBody
+    Iterable<Driver> getDriverWithMostPoints() {
+        return driverRepository.findFirstByOrderByPointsDesc();
+    }
+
+    @GetMapping(path = "/sort-by-wins")
+    public @ResponseBody
+    Iterable<Driver> getDriversSortedByWins() {
+        return driverRepository.findAllByOrderByWinsDesc();
+    }
+
+    @GetMapping(path = "/sort-by-points")
+    public @ResponseBody
+    Iterable<Driver> getDriversSortedByPoints() {
+        return driverRepository.findAllByOrderByPointsDesc();
+    }
+
+    @GetMapping(path = "/wins-above")
+    public @ResponseBody
+    Iterable<Driver> getDriversWithMoreWinsThan(int n) {
+        return driverRepository.findDriversByWinsAfter(n);
+    }
+
+    @GetMapping(path = "/wins-below")
+    public @ResponseBody
+    Iterable<Driver> getDriversWithLessWinsThan(int n) {
+        return driverRepository.findDriversByWinsBefore(n);
+    }
+
+    @DeleteMapping(path = "/del/{id}")
+    public @ResponseBody
+    ResponseEntity<Driver> deleteDriver(@PathVariable Long id) {
+        driverRepository.deleteById(id);
+        return new ResponseEntity<>(HttpStatus.ACCEPTED);
+    }
+
+    @PutMapping(path = "/update/{id}")
+    public ResponseEntity<Driver> updateDriverWins(
+            @PathVariable Long id,
+            @RequestParam Optional<String> firstName,
+            @RequestParam Optional<String> lastName,
+            @RequestParam Optional<Integer> wins,
+            @RequestParam Optional<Integer> poles,
+            @RequestParam Optional<Integer> fastestLaps,
+            @RequestParam Optional<Double> points) {
```

Comment:
This should be an object not just fields

Also no input validation exists. THis is a major problem